stories/color.stories.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import { storiesOf } from '@storybook/react'
+
+import BarChart from '../src/components/bar-chart/'
+import PieChart from '../src/components/pie-chart'
+import LineChart from '../src/components/line-chart'
+import ScatterChart from '../src/components/scatter-chart'
+import barChartData from './data/bar-chart-data'
+import lineChartData from './data/line-chart-data'
+import pieChartData from './data/pie-chart-data'
+import scatterChartData from './data/scatter-chart-data'
+import ResponsiveChartWrapper from './responsive-chart-wrapper'
+
+
+const title = 'My Title'
+const palette = ['#F936A9', '#B54DD5', '#FF4774', '#FFA600', '#1564ec', '#FF7641', '#FFA600']
+
+const colorsMap = {}
+
+storiesOf('Color', module)
+  .add('Bar Chart with Color function', () => {
+    const barIds = ['visits', 'visitors', 'single_visitors', 'repeat_visitors', 'repeat_visitor_rate', 'multi_visitors', 'multi_visitor_rate']
+    barIds.forEach((d, i) => {
+      colorsMap[d] = palette[i]
+    })
+    return (
+      <ResponsiveChartWrapper>
+        <BarChart
+          title={ title }
+          data={ barChartData }
+          axisBottomLegendLabel='Address City'
+          axisLeftLegendLabel='Visitors'
+          groupMode='grouped'
+          colors={(d) => colorsMap[d]}
```

Comment:
Question: how is it different than feeding an array of colors?

Reply:
`colors` param is taken in as a function, so it's referred that way in the component (see around `if (typeof colors === 'function')`)

Reply:
If you just feed a list of colours, the colours would not be all the time the same for Top 5, when you switch tabs. Tamires designed the tabs to always use the same colours for top 5 in overlord. 
Regardless, the nivo charts by themselves could handle colors as functions as default, but Shane added a complex system of setting the default chart-system colours and the default colors as function didn't work anymore, which was practically a bug. Tamires used the function form of colors to set the colours to always match between the charts, so top 5 data in pie clors would always match top 5 in bar chart as colours. The only reason it worked so far, it was because we used only top 5 and Shanes code defaulted to the colour scheme and function in charts, which would do the same thing. Once I did top 10 thought the scheme didn't work anymore, we don't have enough colors in chart-system and the colours were mismatched between the charts.

Reply:
What we do now, we use the colors as function to match specific data attribute value to a specific color, that is how the colors as function works. And as I said, it was not working before, it was a bug that needed to be fixed.

Reply:
I see. I'll probably have to take a look deeper, but from my understanding if nivo is capable of handling colors in both array and function form, it might be able to directly plugged into the nivo chart?

Reply:
Yeah, it might be, but the problem was to keep Shane's process of hiding items in the legend intact, and that's what I did here. I can look over it later if I can make it simpler, although it didn't seem to me when I tried to fix the whole thing.